syft/generic/pointers/pointer_protocol.py

```diff
@@ -92,15 +92,19 @@ def get(self, deregister_ptr: bool = True):
     @staticmethod
     def simplify(worker: AbstractWorker, ptr: "PointerPlan") -> tuple:
 
-        return (ptr.id, ptr.id_at_location, ptr.location.id, ptr.garbage_collect_data)
+        return (
+            ptr.id,
```

Comment:
same remark here. Actually I'm not sure ids should be simplified as they are integers and this only slow down the process

Reply:
Also confused by this.
In some places (e.g. PointerPlan, PointerProtocol, MultiPointerTensor) id is defined as `Union[str, int]` in constructor, in some places (e.g. FixedPrecisionTensor, LoggingTensor) it's mentioned in doc comment that it's str or int. But in parent classes AbstractTensor, AbstractObject, and sy.ID_PROVIDER.pop() that generates those id's, it seems to be just int.

I can make serde to consistently treat ID as int or int/str everywhere, but there might be a reason why it's different for different classes?

Reply:
Just chiming in to say that I have encountered the dual nature of ids as both integers and strings, and I am also confused by this. Would be easier to handle if it were one or the other, but if it needs to be both, would be great to make that consistent (someday and probably not in this PR.)

Reply:
OK, I've made serde follow constructor (or parent constructor) arg type definition or documentation:
torch.nn.Parameter - int
torch.Tensor - int
AdditiveSharingTensor - int or str
FixedPrecisionTensor - int or str
CRTPrecisionTensor - int
LoggingTensor - int or str
MultiPointerTensor - int or str
Plan - int or str
Protocol - int *
PointerTensor - int or str
PointerPlan - int or str
PointerProtocol - int or str
ObjectWrapper - int
ObjectPointer - int or str
TrainConfig - int or str
AutogradTensor - int *
PrivateTensor - int or str
PromiseTensor - int or str

*serde supports int or str, didn't change that